src/app/service.rs

```diff
@@ -47,6 +49,9 @@ impl<T: Send + Sync> Service<http::Request<Incoming>> for AppService<T> {
     type Response = http::Response<ResponseBody>;
 
     fn call(&self, request: http::Request<Incoming>) -> Self::Future {
+        // The middleware stack.
+        let mut deque = VecDeque::new();
```

Comment:
It's probably better to move the `VecDeque` into `Next` as soon as the middleware stack is built.

This way, the only thing that you can do with the middleware stack in user-land is call the next middleware.

Reply:
To reiterate this, at least for the middleware stack–allocation entropy is preferred to preallocating. It's a small allocation that likely comes with zero runtime cost.